docs/getting-started/using-gnokey.md

```diff
@@ -0,0 +1,777 @@
+
+## Interacting with a Gno.land chain
+
+`gnokey` allows you to interact with any Gno.land network, such as the
+[Portal Loop](../concepts/portal-loop.md) testnet.
+
+There are multiple ways anyone can interact with the chain:
+- Transactions - state-changing calls which use gas
+- ABCI queries - read-only calls which do not use gas
+
+Both transactions and ABCI queries can be made via `gnokey`'s subcommands,
+`maketx` and `query`.
+
+## State-changing calls (transactions)
+
+In Gno, there are four types of messages that can change on-chain state:
+- `AddPackage` - adds new code to the chain
+- `Call` - calls a specific path and function on the chain
+- `Send` - sends coins from one address to another
+- `Run` - executes a Gno script against on-chain code
+
+A Gno.land transaction contains two main things:
+- A base configuration where variables such as `gas-fee`, `gas-wanted`, and others
+  are defined
+- A list of messages to execute on the chain
+
+Currently, `gnokey` supports single-message transactions, while multiple-message
+transactions can be created in Go programs, supported by the
+[gnoclient](../reference/gnoclient/gnoclient.md) package.
+
+We will need some testnet coins (GNOTs) for each state-changing call. Visit the [Faucet
+Hub](https://faucet.gno.land) to get GNOTs for the Gno testnets that are currently live.
+
+Let's delve deeper into each of these message types.
+
+### `AddPackage`
+
+In case you want to upload new code to the chain, you can use the `AddPackage`
+message type. You can send an `AddPackage` transaction with `gnokey` using the
+following command:
+
+```bash
+gnokey maketx addpkg
+```
+
+To understand how to use this subcommand better, let's write a simple "Hello world"
+[pure package](../concepts/packages.md). First, let's create a folder which will
+store our example code.
+
+```bash
+└── example/
+```
+
+Then, let's create a `hello_world.gno` file under the `p/` folder:
+
+```bash
+cd example
+mkdir p/ && cd p
+touch hello_world.gno
+```
+
+Now, we should have the following folder structure:
+
+```bash
+└── example/
+│   └── p/
+│       └── hello_world.gno
+```
+
+In the `hello_world.gno` file, add the following code:
+
+```go
+package hello_world
+
+func Hello() string {
+  return "Hello, world!"
+}
+```
+
+We are now ready to upload this package to the chain. To do this, we must set the
+correct flags for the `addpkg` subcommand.
+
+The `addpkg` subcommmand uses the following flags and arguments:
+- `-pkgpath` - on-chain path where your code will be uploaded to
+- `-pkgdir` - local path where your is located
+- `-broadcast` - enables broadcasting the transaction to the chain
+- `-send` - a deposit amount of GNOT to send along with the transaction
+- `-gas-wanted` - the upper limit for units of gas for the execution of the
+  transaction
+- `-gas-fee` - amount of GNOTs to pay per gas unit
+- `-chain-id` - id of the chain to connect to
```

Comment:
We don't use this chain id for connecting

Reply:
Updated it to be more accurate: [78187b0](https://github.com/gnolang/gno/pull/2226/commits/78187b0629a86f6fae716e358191202a5f709efa)